### PR TITLE
Correctly handle functors when auto-generating column names

### PR DESCRIPTION
**Pull request by bkamins**
An interesting corner-case fix of:
```
julia> struct Identity
       end

julia> function (::Identity)(x)
           return x
       end

julia> i = Identity()
Identity()

julia> df = DataFrame(x=1:3)
3×1 DataFrame
 Row │ x
     │ Int64
─────┼───────
   1 │     1
   2 │     2
   3 │     3

julia> combine(df, :x => identity∘i)
ERROR: MethodError: no method matching nameof(::Identity)
Closest candidates are:
  nameof(::Module) at reflection.jl:16
  nameof(::DataType) at reflection.jl:214
  nameof(::UnionAll) at reflection.jl:215
  ...
Stacktrace:
 [1] funname(f::Identity)
```

### CODE REVIEW

**Comment by bkamins** [comment]
Nightly fails because of https://github.com/JuliaLang/julia/pull/42991.

File: src/other/utils.jl [hunk 1]
```diff
@@ -120,8 +120,13 @@ function gennames(n::Integer)
 end
 
 function funname(f)
-    n = nameof(f)
-    String(n)[1] == '#' ? :function : n
+    try
```
Comment (nalimilan) [inline src/other/utils.jl:123]: Maybe better use `applicable` rather than `try`?

Comment (bkamins) [reply src/other/utils.jl]: ok - changed

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**